lib/3scale_toolbox/commands/remote_command/remote_add.rb

```diff
@@ -0,0 +1,75 @@
+require 'cri'
+require '3scale/api'
+require '3scale_toolbox/base_command'
+
+module ThreeScaleToolbox
+  module Commands
+    module RemoteCommand
+      class RemoteAddSubcommand < Cri::CommandRunner
+        include ThreeScaleToolbox::Command
+        def self.command
+          Cri::Command.define do
+            name        'add'
+            usage       'add <remote_name> <remote_url>'
+            summary     '3scale CLI remote add'
+            description '3scale CLI command to add new remote'
+            param       :remote_name
+            param       :remote_url
+            runner RemoteAddSubcommand
+          end
+        end
+
+        def run
+          # 'arguments' cannot be converted to Hash
+          add_remote arguments[:remote_name], arguments[:remote_url]
+        rescue StandardError => e
+          warn e.message
+          warn e.backtrace
+          exit 1
+        end
+
+        def validate_remote_name(name)
+          remotes = config.data :remotes
```

Comment:
Wouldn't it be better to actually return an empty Hash instead of nil? That would reduce the defensive programming you have to do in few places.

Reply:
And enforce Configuration values to be Hash objects?

Reply:
I personally prefer Structs or own classes, but that works too.

Reply:
I have implemented `ThreeScaleToolbox::Remotes` module to abstract all operations regarding remotes. It includes some validations and parsing logic, besides accessing/updating remote data from configuration.

WDYT?

Reply:
That's better! I personally prefer classes over modules, but it is still an improvement 👍